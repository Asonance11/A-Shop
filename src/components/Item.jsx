import React from 'react';
import { Link } from 'react-router-dom';

const Item = (props) => {
	const product = props.product;
	return (
		<section className="w-full flex flex-col gap-2">
			<img src={product.image} alt={product.alt} className="w-full" />
			<hr />
			<ul className="text-white flex flex-col gap-4">
				<li className="flex items-center justify-between">
					<p>Photographer</p>
					<p>{product.photographer}</p>
				</li>
				<li className="flex items-center justify-between">
					<p>Price</p>
					<p>{`$${product.price}`}</p>
				</li>
				<li className="flex items-center justify-between">
					<button
						type="button"
						className="py-2 px-4 bg-gradient-to-r from-forestGreen via-adventurousYellow to-oceanTurquoise rounded-md text-darkGrey"
					>
						<Link to={`/shop/${product.id}`}>More Information</Link>
					</button>
				</li>
			</ul>
		</section>
	);
};

export default Item;
