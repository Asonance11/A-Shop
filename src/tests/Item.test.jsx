import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, it } from 'vitest';
import Item from '../components/Item';

describe('Item component', () => {
	it('renders in the Shop Page', () => {
		const anItem = {
			image: 'www.awebsite.com/something.jpeg',
			alt: 'An image',
			photographer: 'me',
			price: 10000,
		};

		const { container } = render(<Item product={anItem} />, {
			wrapper: MemoryRouter,
		});
		expect(container).toMatchSnapshot();
	});
});
