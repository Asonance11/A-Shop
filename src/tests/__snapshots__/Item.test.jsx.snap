// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Item component > renders in the Shop Page 1`] = `
<div>
  <section
    class="w-full flex flex-col gap-2"
  >
    <img
      alt="An image"
      class="w-full"
      src="www.awebsite.com/something.jpeg"
    />
    <hr />
    <ul
      class="text-white flex flex-col gap-4 font-sans"
    >
      <li
        class="flex items-center justify-between"
      >
        <p>
          Photographer
        </p>
        <p>
          me
        </p>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <p>
          Price
        </p>
        <p>
          $10000
        </p>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <button
          class="py-2 px-4 bg-gradient-to-r from-forestGreen via-adventurousYellow to-oceanTurquoise rounded-md text-darkGrey transition-all duration-500 ease-in-out   hover:from-foggyBlue hover:to-forestGreen"
          type="button"
        >
          <a
            href="/shop/undefined"
          >
            More Information
          </a>
        </button>
      </li>
    </ul>
  </section>
</div>
`;
